Sprint 7 solution in file manager

Omorlan opened the pull request:
Доработана функциональность по техническому заданию спринта

Omorlan committed:
feat: Добавить новый тип менеджера

Omorlan committed:
feat: Добавить новые тесты
feat: Изменить менеджер по умолчанию
fix: Убрать лишние импорты
fix: Исправить ошибку удаления из истории не всех сабтасков при удалении эпика
fix: исправлена ошибка создания новых узлов в списке истории

johnskyi commented on src/sprint4/managers/filemanager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,251 @@
+        this.file = file;
+    }
+
+    public void save() {
```
Этот метод лучше сделать приватным. Его нигде кроме внутри этого класса не применяем

Omorlan replied:
Поправил

johnskyi commented on src/sprint4/managers/filemanager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,251 @@
+
+    @Override
+    public Task getTaskById(int id) {
+        save();
```
ты сохраняешь раньше чем возвращаешь таску, так сохранение идет некорректно. Делай так Task task = super.getTaskById(); save(); return task;  только в таком порядке. Поправь пожалуйста все похожие места

Omorlan replied:
Поправил

johnskyi commented on src/sprint4/managers/filemanager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,251 @@
+    }
+
+    @Override
+    public List<SubTask> getSubTasksOfEpic(Epic epic) {
```
здесь аналогично верхнему комментарию

johnskyi commented on test/src/managers/filemanager/FileBackedTaskManagerTest.java:
```diff
@@ -0,0 +1,48 @@
+
+
+    @Test
+    public void SaveAndLoadFromFileMustBeSame() throws IOException {
```
название метода с маленькой буквы и маловато тестов дял менеджера, нужно протестить все случаи и весь функционал

Omorlan replied:
Добавил, 94% строчек покрыто учитывая тесты для родительского класса

Omorlan committed:
fix: Исправить ошибку при работе с историей
fix: Исправить поведение сохранения при вызове методов
fix: Добавить дополнительные тесты

Omorlan committed:
fix:CodeStyle

Omorlan committed:
fix:CodeStyle

Omorlan committed:
fix:CodeStyle

Omorlan committed:
fix:CodeStyle
